crates/flox-rust-sdk/src/models/environment.rs

```diff
@@ -33,75 +34,91 @@ pub enum InstalledPackage {
     StorePath(StorePath),
 }
 
-/// Struct representing a local environment in a given location and state
+#[async_trait(?Send)]
+pub trait Environment {
+    type ConcreteTemporary;
```

Comment:
@zmitchell I chose to do it this way because `replace_with` needs to use the same type as `modify_in`

Reply:
I'm not really sure what this comment is referring to

Reply:
Instead of returning `Box<dyn Temporary>` in `replace_with`, I added this type